misc/releaser.py

```diff
@@ -363,39 +368,63 @@ def collect_newsfragments() -> list[Path]:
     return fragments
 
 
-def build_release(version: Version, yes: bool) -> None:
-    if version.is_dev:
-        raise ReleaseError(f"Don't release a dev version: {version}")
+def get_release_branch(version: Version) -> str:
+    return f"releases/{version.major}.{version.minor}"
 
-    print(f"Building release {COLOR_GREEN}{version}{COLOR_END} ...")
-    old_version = get_version_from_code()
+
+def ensure_current_code_version_is_smaller(version: Version) -> Version:
+    old_version: Version = get_version_from_code()
     if version <= old_version:
         raise ReleaseError(
             f"Previous version is greater that the new version ({COLOR_YELLOW}{old_version}{COLOR_END} >= {COLOR_YELLOW}{version}{COLOR_END})"
         )
 
-    now = datetime.utcnow()
-    release_date = now.date()
-    # Cannot just add years to date given it wouldn't handle february 29th
-    license_conversion_date = datetime.fromtimestamp(
-        now.timestamp() + LICENSE_CONVERSION_DELAY
-    ).date()
-    assert release_date.toordinal() < license_conversion_date.toordinal()
+    return old_version
 
-    # Check repo is clean
+
+def ensure_working_in_a_clean_git_repo() -> None:
     stdout = run_git("status", "--porcelain", "--untracked-files=no")
     if stdout.strip():
         raise ReleaseError("Repository is not clean, aborting")
 
-    # Update BUSL license date marker & version info
-    update_license_file(version, license_conversion_date)
-    # Update parsec version
-    update_version_files(version)
 
-    # Update HISTORY.rst
+def ensure_working_on_the_correct_branch(
+    release_branch: str, base_ref: str | None, version: Version
+) -> None:
+    current_branch = run_git("rev-parse", "--abbrev-ref", "HEAD")
+    print(f"Current branch {COLOR_GREEN}{current_branch}{COLOR_END}")
+
+    if current_branch == release_branch:
+        print("Already in the release branch, nothing to do.")
+        return
+
+    if version.patch == 0:
+        print("Creating release branch ...")
+        run_git("switch", "--create", release_branch, *([base_ref] if base_ref else []))
+    else:
+        raise ReleaseError(
+            f"""
+        You are trying to create a patched release but you are using the wrong base branch.
+        The base branch should be `{release_branch}`.
+        (you should already have the commit related to the patch pushed to the branch)
+        """
+        )
+
+
+def get_release_date_and_licence_eol() -> tuple[date, date]:
+    now = datetime.utcnow()
+    release_date = now.date()
+    # Cannot just add years to date given it wouldn't handle february 29th
+    license_eol_date = datetime.fromtimestamp(now.timestamp() + LICENSE_CONVERSION_DELAY).date()
+    assert release_date.toordinal() < license_eol_date.toordinal()
+    return release_date, license_eol_date
```

Comment:
Better to avoid "and" in function names (code smell), functions should only do one thing.

I would:
- move the `release_date` out of the function (it is only one line)
- rename the function to only retrieve a valid `licence_eol_date`

```suggestion
def get_licence_eol_date(release_date: date) -> date:
    now = datetime.utcnow()
    # Cannot just add years to date given it wouldn't handle february 29th
    license_eol_date = datetime.fromtimestamp(now.timestamp() + LICENSE_CONVERSION_DELAY).date()
    assert release_date.toordinal() < license_eol_date.toordinal()
    return license_eol_date
```

Reply:
done